operator/README.md

```diff
@@ -1,2 +1,92 @@
-# pmem-csi-operator
-Operator for PMEM-CSI
+# PMEM-CSI-Operator
+
+Kubernetes operator for [PMEM-CSI](https://github.com/intel/pmem-csi) driver deployment. Using this operator one can easily deploy and manage PMEM-CSI driver in a Kubernetes cluster.
+
+The PMEM-CSI operator is based on the CoreOS [operator-sdk](https://github.com/operator-framework/operator-sdk) tools and APIs.
+
+## PMEM-CSI Deployment custom resource definition(CRD)
+
+Current PMEM-CSI Deployment object supports below API:
+
+### Deployment
+
+|Field | type | Description |
+|---|---|---|
+| apiVersion | string  | `pmem-csi.intel.com/v1alpha1`|
+| kind | string | `Deployment`|
+| metaData | [ObjectMeta](https://git.k8s.io/community/contributors/devel/api-conventions.md#spec-and-status) | Standard objects metadata. |
+| spec | [DeploymentSpec](#deployment-spec) | Specification of the desired behavior of the deployment |
+
+### DeployemtSpec
+
+|Field | type | Description | Default Value |
+|---|---|---|---|
+| image | ImageInfo | PMEM-CSI driver container image information | `intel/pmem-csi-driver:canary` |
```

Comment:
> If I do a `go get -o yaml deployment.pmem-csi.intel.com/v1alpha1`, will it show what the default is or only the explicitly set values?

You mean `kubectl get -o yaml deployment.pmem-csi.intel.com example-deployment`?
No, it does not show the defaults, it only shows the values set by the user as we do not update the Deployment object. Instead, we prepare(/update?) the secondary resources using the `DeploymentSpec` generated with the combined values(defaults + user set).

Do you feel we should update the Deployment itself with the defaults we used?

Reply:
> I'm not sure how that is normally handled. What don't want is that defaults are stored as if the user had set them intentionally. For example, if we change the default here from "intel/pmem-csi-driver:v0.6.0" to "intel/pmem-csi-driver:v0.6.1", then a deployment where the user hasn't explicitly selected the older version should get updated,

If we consider this, to make this work smoothly, I guess we should expose these defaults as Operator configuration, say `ConfigMap`. So that admin alters the defaults by updating ConfigMap, then the Operator catches these changes and redeploys the driver for existing Deployment(s).

Reply:
> Do you feel we should update the Deployment itself with the defaults we used?

No, not at all ("What don't want is that defaults are stored as if the user had set them intentionally."). We must be able to distinguish between behavior chosen by the user and behavior selected because it is the default. Then the default can change.

> If we consider this, to make this work smoothly, I guess we should expose these defaults as Operator configuration, say ConfigMap.

I don't think we need this. The defaults can be built into the operator. One usage pattern that was mentioned for OpenShift/Operator Hub is that an admin opts into auto-updating operators. Then the updated defaults for versions also cause the corresponding software to be updated *without* the admin being involved.

My question was just whether those defaults then can be observed somehow or must be documented.